tests/transforms/test_experimental/test_transform_dispatcher.py

```diff
@@ -179,6 +190,42 @@ def qnode_circuit(a):
         )
         assert not dispatched_transform.is_informative
 
+    def test_integration_dispatcher_with_informative_transform(self):
+        """Test that no error is raised with the transform function and that the transform dispatcher returns
+        the right object when an informative transform is applied."""
+
+        dispatched_transform = transform(informative_transform, is_informative=True)
+
+        # Applied on a tape (return processed results)
+        expected = len(tape_circuit.operations)
+        num_ops = dispatched_transform(tape_circuit)
+        assert num_ops == expected
+
+        # Applied on a qfunc (return a qfunc)
+        qfunc = dispatched_transform(qfunc_circuit)
+        assert callable(qfunc)
+
+        # Applied on a qnode (return a qnode with populated the program)
+        @qml.qnode(device=dev)
+        def qnode_circuit(a):
+            """QNode circuit."""
+            qml.Hadamard(wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.PauliX(wires=0)
+            qml.RZ(a, wires=1)
+            return qml.expval(qml.PauliZ(wires=0))
+
+        qnode_transformed = dispatched_transform(qnode_circuit)
+        assert not qnode_circuit.transform_program
+
+        assert qnode_transformed(0.1) == 4
+        assert isinstance(qnode_transformed, qml.QNode)
```

Comment:
idk why but it feels a little strange to me that informative QNodes are still QNodes. I don't think I have any alternative ideas so I'll leave it, but maybe some users will be surprised like me?

Reply:
Yes, because we need to store the transform program since users can apply other transforms as well.